[BUG] single-bit std_logic to a slice

- [X] I have verified this bug has not already been reported on the [upstream issue tracker](https://github.com/SystemRDL/PeakRDL-regblock/issues)
- [X] I have reviewed this project's [contribution guidelines](https://github.com/SystemRDL/PeakRDL-regblock-vhdl/blob/main/CONTRIBUTING.md)

**Describe the bug**
peakrdl-regblock-vhdl used (0 => value) when assigning single-bit std_logic to a slice, instead of (low => value), causing “choice is out of index range” when low != 0

This was found in `test_wide_regs` which is present in this repo.  I am not sure what compiler you are using, but maybe not ghdl (?) but this is what I got, just during compilation. 

    reg {
        regwidth = 64;
        accesswidth = 16;
        default sw=rw;
        default hw=r;

        field {} f1[7:0] = 0;
        field {} f2[14:12] = 0;
        field {} f3[36:36] = 0;
        field {} f4[47:40] = 0;
    } rw_reg1;


Details like these can be helpful:
```
ghdl -i --std=08  --workdir=sim_build --work=work ./regblock-vhdl-rtl/*.vhd ../interfaces/reg_utils.vhd && \
ghdl -m --std=08  --workdir=sim_build -Psim_build --work=work regblock_wrapper
./regblock-vhdl-rtl/regblock.vhd:891:41:error: choice is out of index range
    readback_array(1)(4 downto 4) <= (0 => field_storage.rw_reg1.f3.value) when (decoded_reg_strb.rw_reg1(2) and not decoded_req_is_wr) else (others => '0');
                                        ^
./regblock-vhdl-rtl/regblock.vhd:902:41:error: choice is out of index range
    readback_array(5)(4 downto 4) <= (0 => field_storage.rw_reg1_lsb0.f3.value) when (decoded_reg_strb.rw_reg1_lsb0(2) and not decoded_req_is_wr) else (others => '0');
                                        ^

```


**Expected behavior**
A clear and concise description of what you expected to happen.


**Additional context**
Add any other context about the problem here.

--- a/src/peakrdl_regblock_vhdl/readback/generators.py
+++ b/src/peakrdl_regblock_vhdl/readback/generators.py
@@ -248,7 +248,7 @@ class ReadbackAssignmentGenerator(RDLForLoopGenerator):
 
                 if field.width == 1:
                     # convert from std_logic to std_logic_vector
-                    value = f"({field.low} => {value})"
+                    value = f"to_std_logic_vector({value})"
                 self.add_content(f"readback_array({self.current_offset_str})({field.high} downto {field.low}) <= {value} when {rd_strb} else (others => '0');")
                 bidx = field.high + 1
 
@@ -317,7 +317,7 @@ class ReadbackAssignmentGenerator(RDLForLoopGenerator):
 
                     if field.width == 1:
                         # convert from std_logic to std_logic_vector
-                        value = f"({low} => {value})"
+                        value = f"to_std_logic_vector({value})"
                     self.add_content(f"readback_array({self.current_offset_str})({high} downto {low}) <= {value} when {rd_strb} else (others => '0');")
 
                     current_bit = field.high + 1

